js/src/components/tree-select-control/index.js

```diff
@@ -169,30 +194,104 @@ const TreeSelectControl = ( {
 			);
 		};
 
-		const filterOption = ( option ) => {
-			if ( hasChildren( option ) ) {
-				option.children = option.children.filter( filterOption );
-				return !! option.children.length;
-			}
+		const descriptors = {
+			hasChildren: {
+				get() {
+					return this.children?.length > 0;
+				},
+			},
```

Comment:
💅  Missing docs

Reply:
Added in https://github.com/woocommerce/google-listings-and-ads/pull/1522/commits/6dfadfc6f50e9938f0f5434a1b4fdad1a9bcf93c.